RecoCTPPS/PixelLocal/plugins/CTPPSPixelClusterProducer.cc

```diff
@@ -0,0 +1,77 @@
+
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "RecoCTPPS/PixelLocal/interface/CTPPSPixelClusterProducer.h"
+
+CTPPSPixelClusterProducer::CTPPSPixelClusterProducer(const edm::ParameterSet& conf) :
+  param_(conf) ,
+  clusterizer_(conf){
+  
+  src_ = conf.getParameter<std::string>("label");
+  verbosity_ = conf.getUntrackedParameter<int> ("RPixVerbosity");
+	 
+  tokenCTPPSPixelDigi_ = consumes<edm::DetSetVector<CTPPSPixelDigi> >(edm::InputTag(src_));
+ 
+  produces<edm::DetSetVector<CTPPSPixelCluster> > ();
+  }
+
+CTPPSPixelClusterProducer::~CTPPSPixelClusterProducer() {
+
+}
+
+void CTPPSPixelClusterProducer::fillDescriptions(edm::ConfigurationDescriptions & descriptions){
+  edm::ParameterSetDescription desc;
+  desc.addUntracked<int>("RPixVerbosity",0);
+  desc.add<std::string>("label", "ctppsPixelDigis");
+  desc.add<int>("SeedADCThreshold",15);
+  desc.add<int>("ADCThreshold",10);
+  desc.add<double>("ElectronADCGain",135.0);
+  desc.add<int>("VCaltoElectronGain",50);
+  desc.add<int>("VCaltoElectronOffset",-411);
+  desc.add<bool>("doSingleCalibration",false); 
+ descriptions.add("ctppsPixelClusters", desc);
+}
+
+void CTPPSPixelClusterProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup) {
+  
+/// get inputs
+  edm::Handle<edm::DetSetVector<CTPPSPixelDigi> > rpd;
+  iEvent.getByToken(tokenCTPPSPixelDigi_, rpd);
+ 
+// get analysis mask to mask channels
+  edm::ESHandle<CTPPSPixelAnalysisMask> aMask;
+
+  if(rpd->size())
+  iSetup.get<CTPPSPixelAnalysisMaskRcd>().get("RPix",aMask);
+  
```

Comment:
the label in this line  is the origin of the recent failures in the add on tests.

Questions / comments:

. this piece of code expects a label which has not been specified when the tags were submitted to the queues to make global tags. As such, the request of conditions cannot be met when CTPPSPixelClusterProducer is run

. we're not aware that there need to be multiple instances of CTPPSPixelAnalysisMaskRcd to be consumed from GT; in that case, no label is necessary; if that's so, please remove the specification of the label from iSetup.get

.